src/Middleware/ParametersNameDecodingHandler.php

```diff
@@ -75,9 +75,22 @@ private function decodeQueryParameters(RequestInterface $request): RequestInterf
         if (!$this->decodingOption->isEnabled() || !$this->decodingOption->getParametersToDecode()) {
             return $request;
         }
-        $encodingsToReplace = array_map(function ($character) { return "%".dechex(ord($character)); }, $this->decodingOption->getParametersToDecode());
-        /** @var string $decodedUri */
-        $decodedUri = str_ireplace($encodingsToReplace, $this->decodingOption->getParametersToDecode(), $request->getUri());
+        $decodedUri = self::decodeUriEncodedString($request->getUri(), $this->decodingOption->getParametersToDecode());
         return $request->withUri(new Uri($decodedUri));
     }
+
```

Comment:
We could add the `com.microsoft.kiota.handler...enable` attribute value for this and other handlers